folly/experimental/settings/Types.cpp

```diff
@@ -27,6 +29,8 @@ std::string_view toString(SetErrorCode code) {
       return "rejected";
     case SetErrorCode::FrozenImmutable:
       return "frozen immutable";
+    default:
+      throw std::invalid_argument("Code '" + std::to_string(static_cast<int>(code)) + "' had no string representation!");
```

Comment:
IMO this should not be an error - it's for diagnostics.

We should like to return `"(unknown)"` in case of an unrecognized input. But we should also like the compiler to warn us when the `case`s in the `switch` are non-exhaustive.